compiler/rustc_typeck/src/check/writeback.rs

```diff
@@ -352,9 +352,9 @@ impl<'cx, 'tcx> WritebackCx<'cx, 'tcx> {
         assert_eq!(fcx_typeck_results.hir_owner, self.typeck_results.hir_owner);
         let common_hir_owner = fcx_typeck_results.hir_owner;
 
-        for (&id, &origin) in fcx_typeck_results.closure_kind_origins().iter() {
-            let hir_id = hir::HirId { owner: common_hir_owner, local_id: id };
-            self.typeck_results.closure_kind_origins_mut().insert(hir_id, origin);
+        for (id, origin) in fcx_typeck_results.closure_kind_origins().iter() {
```

Comment:
`into_iter()` and then move origin?

Reply:
Correct me if I am wrong, but in order to use into_iter we would need to implement `IntoIterator` and `Iterator`

```
error[E0599]: no method named `into_iter` found for struct `ty::context::LocalTableInContext<'_, (rustc_span::Span, rustc_middle::hir::place::Place<'_>)>` in the current scope
   --> compiler/rustc_typeck/src/check/writeback.rs:387:71
    |
387 |         for (id, origin) in fcx_typeck_results.closure_kind_origins().into_iter() {
    |                                                                       ^^^^^^^^^ method not found in `ty::context::LocalTableInContext<'_, (rustc_span::Span, rustc_middle::hir::place::Place<'_>)>`
    |
   ::: /users/rlmmfruy/rust/compiler/rustc_middle/src/ty/context.rs:197:1
    |
197 | pub struct LocalTableInContext<'a, V> {
    | -------------------------------------
    | |
    | doesn't satisfy `_: IntoIterator`
    | doesn't satisfy `_: Iterator`
    |
    = note: the method `into_iter` exists but the following trait bounds were not satisfied:
            `ty::context::LocalTableInContext<'_, (rustc_span::Span, rustc_middle::hir::place::Place<'_>)>: Iterator`
            which is required by `ty::context::LocalTableInContext<'_, (rustc_span::Span, rustc_middle::hir::place::Place<'_>)>: IntoIterator`
            `&ty::context::LocalTableInContext<'_, (rustc_span::Span, rustc_middle::hir::place::Place<'_>)>: Iterator`
            which is required by `&ty::context::LocalTableInContext<'_, (rustc_span::Span, rustc_middle::hir::place::Place<'_>)>: IntoIterator`
            `&mut ty::context::LocalTableInContext<'_, (rustc_span::Span, rustc_middle::hir::place::Place<'_>)>: Iterator`
            which is required by `&mut ty::context::LocalTableInContext<'_, (rustc_span::Span, rustc_middle::hir::place::Place<'_>)>: IntoIterator`
            ```

Reply:
@arora-aman Let me know if you want me to still do that or not

Reply:
Talked with Aman and decided to keep `iter()`